src/category_theory/monoidal/category.lean

```diff
@@ -0,0 +1,384 @@
+-- right unitor:
+(right_unitor             : Π X : C, X ⊗ tensor_unit ≅ X)
+(right_unitor_naturality' :
+  right_unitor_natural tensor_obj @tensor_hom tensor_unit right_unitor . obviously)
+-- pentagon identity:
+(pentagon'                : pentagon @tensor_hom associator . obviously)
+-- triangle identity:
+(triangle'                :
+  triangle @tensor_hom left_unitor right_unitor associator . obviously)
+
+restate_axiom monoidal_category.tensor_id'
+attribute [simp] monoidal_category.tensor_id
+restate_axiom monoidal_category.tensor_comp'
+attribute [simp] monoidal_category.tensor_comp
+restate_axiom monoidal_category.associator_naturality'
+restate_axiom monoidal_category.left_unitor_naturality'
+restate_axiom monoidal_category.right_unitor_naturality'
+restate_axiom monoidal_category.pentagon'
+restate_axiom monoidal_category.triangle'
+attribute [simp] monoidal_category.triangle
+
+open monoidal_category
+
+infixr ` ⊗ `:70 := tensor_obj
+infixr ` ⊗ `:70 := tensor_hom
+
+local notation `𝟙_` := tensor_unit
+local notation `α_` := associator
+local notation `λ_` := left_unitor
+local notation `ρ_` := right_unitor
+
+def tensor_iso {C : Sort u} {X Y X' Y' : C} [monoidal_category.{v} C] (f : X ≅ Y) (g : X' ≅ Y') :
+    X ⊗ X' ≅ Y ⊗ Y' :=
+{ hom := f.hom ⊗ g.hom,
+  inv := f.inv ⊗ g.inv,
+  hom_inv_id' := by rw [←tensor_comp, iso.hom_inv_id, iso.hom_inv_id, ←tensor_id],
+  inv_hom_id' := by rw [←tensor_comp, iso.inv_hom_id, iso.inv_hom_id, ←tensor_id] }
+
+infixr ` ⊗ `:70 := tensor_iso
+
+namespace monoidal_category
+
+section
+
+variables {C : Sort u} [𝒞 : monoidal_category.{v} C]
+include 𝒞
+
+instance tensor_is_iso {W X Y Z : C} (f : W ⟶ X) [is_iso f] (g : Y ⟶ Z) [is_iso g] : is_iso (f ⊗ g) :=
+{ ..(as_iso f ⊗ as_iso g) }
+
+@[simp] lemma inv_tensor {W X Y Z : C} (f : W ⟶ X) [is_iso f] (g : Y ⟶ Z) [is_iso g] :
+  inv (f ⊗ g) = inv f ⊗ inv g := rfl
+
+variables {U V W X Y Z : C}
+
+-- When `rewrite_search` lands, add @[search] attributes to
+
+-- monoidal_category.tensor_id monoidal_category.tensor_comp monoidal_category.associator_naturality
+-- monoidal_category.left_unitor_naturality monoidal_category.right_unitor_naturality
+-- monoidal_category.pentagon monoidal_category.triangle
+
+-- tensor_comp_id tensor_id_comp comp_id_tensor_tensor_id
+-- triangle_assoc_comp_left triangle_assoc_comp_right triangle_assoc_comp_left_inv triangle_assoc_comp_right_inv
+-- left_unitor_tensor left_unitor_tensor_inv
+-- right_unitor_tensor right_unitor_tensor_inv
+-- pentagon_inv
+-- associator_inv_naturality
+-- left_unitor_inv_naturality
+-- right_unitor_inv_naturality
+
+@[simp] lemma comp_tensor_id (f : W ⟶ X) (g : X ⟶ Y) :
+  (f ≫ g) ⊗ (𝟙 Z) = (f ⊗ (𝟙 Z)) ≫ (g ⊗ (𝟙 Z)) :=
+by { rw ←tensor_comp, simp }
+
+@[simp] lemma id_tensor_comp (f : W ⟶ X) (g : X ⟶ Y) :
+  (𝟙 Z) ⊗ (f ≫ g) = (𝟙 Z ⊗ f) ≫ (𝟙 Z ⊗ g) :=
+by { rw ←tensor_comp, simp }
+
+lemma id_tensor_comp_tensor_id (f : W ⟶ X) (g : Y ⟶ Z) :
```

Comment:
Just wondering, do you actually use this lemma elsewhere? It's not a simp lemma and it doesn't seem to appear elsewhere in this PR. I would have guessed that simplifying to `g tensor f` would usually be more useful.

Reply:
I actually don't remember if I wrote this or @mjendrusch did. If it was me, it was because this "is" an axiom in Vicary's axiomatisation of 2-categories, and I eventually want to make contact with that. 

In any case, as you point out, the pair of `simp` lemmas is more useful, and I've replaced this with the pair.